apps/files/src/components/PublicLinksSidebar/EditPublicLink.vue

```diff
@@ -0,0 +1,292 @@
+          <oc-button :disabled="!$_isValid" @click="$_addLink" v-if="!linksLoading && $_isNew" variation="primary"><translate>Create</translate></oc-button>
+          <oc-button :disabled="!$_isValid || !$_hasChanges" @click="$_updateLink" v-else-if="!linksLoading && !$_isNew" variation="primary"><translate>Save</translate></oc-button>
+          <button class="uk-button uk-button-default uk-position-relative" disabled v-else>
+            <template v-if="$_isNew">
+              <oc-spinner :ariaLabel="$gettext('Creating Public Link')" class="uk-position-small uk-position-center-left" size="small"/>
+              <span :aria-hidden="true" class="uk-margin-small-left" v-translate>Creating Public Link</span>
+            </template>
+            <template v-else>
+              <oc-spinner :ariaLabel="$gettext('Saving Public Link')" class="uk-position-small uk-position-center-left" size="small"/>
+              <span :aria-hidden="true" class="uk-margin-small-left" v-translate>Saving Public Link</span>
+            </template>
+          </button>
+        </div>
+      </oc-grid>
+    </form>
+  </div>
+</template>
+<script>
+import { mapGetters, mapActions } from 'vuex'
+import mixins from '../../mixins'
+import moment from 'moment'
+import filter from 'lodash/filter'
+import first from 'lodash/first'
+import values from 'lodash/values'
+import publicLinkRoles from '../../helpers/publicLinkRolesDefinition'
+const RolesSelect = () => import('../Roles/RolesSelect.vue')
+
+export default {
+  mixins: [mixins],
+  props: ['params'],
+  components: {
+    RolesSelect
+  },
+  data () {
+    return {
+      password: null,
+      errors: false,
+      ...this.params,
+
+      placeholder: {
+        expireDate: this.$gettext('Expiration date'),
+        password: this.$gettext('Password'),
+        mailTo: this.$gettext('Mail recipients'),
+        mailBody: this.$gettext('Personal note')
+      }
+    }
+  },
+  title: ($gettext) => {
+    return $gettext('Links')
+  },
+  computed: {
+    ...mapGetters('Files', ['highlightedFile', 'linksLoading']),
+    ...mapGetters(['getToken', 'capabilities']),
+
+    $_isNew () {
+      return !this.params.id
+    },
+
+    $_isFolder () {
+      return this.highlightedFile.type === 'folder'
+    },
+
+    $_isFile () {
+      return !this.$_isFolder
+    },
+
+    $_hasChanges () {
+      return (this.expireDate !== this.params.expireDate) ||
+          (this.name !== this.params.name) ||
+          (this.permissions !== this.params.permissions) ||
+          (this.password !== null)
+    },
+
+    $_sendMailEnabled () {
+      return Object.keys(this.capabilities.files_sharing.public.send_mail).length > 0
+    },
+
+    $_roles () {
+      const $gettext = this.$gettext
+      return publicLinkRoles({
+        $gettext,
+        isFolder: this.$_isFolder
+      })
+    },
+
+    $_selectedRole () {
+      const permissions = parseInt(this.permissions)
```

Comment:
always use `parseInt(x, 10)` to be safer, because leading zeroes would have it parsed as octal

Reply:
👍

Reply:
Hey wait a second, Isn't octal deprecated? Afaik only a `0x` will involve magic (base 16), everything else is treated as base 10.

Reply:
not sure, this is an old habit I've been carrying over the years...

Reply:
Still safer with explicit base. Added. ;-)